### PR TITLE
Grounding DINO enhancements and bugfixes

### PR DESCRIPTION
**Pull request by skylargivens**
# Description

- Allows users to pass custom `box_threshold` and `text_threshold` params to Grounding DINO core model.
- Update docs to reflect `box_threshold` and `text_threshold` params.
- Fixes error by filtering out detections where text similarity is lower than `text_threshold` and Grounding DINO returns `None` for class ID.
- Fixes images passed to Grounding DINO model being loaded as RBG instead of BGR.
- Adds NMS to Grounding DINO, optionally using class agnostic NMS via `CLASS_AGNOSTIC_NMS` env var.

## Type of change

Please delete options that are not relevant.

-   ✅  Bug fix (non-breaking change which fixes an issue)
-   ✅  New feature (non-breaking change which adds functionality)
-   ✅  This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

- Tested Grounding DINO inference API with and without new `box_threshold` and `text_threshold` params.
- Tested that detections filtered out by text threshold no longer cause an error (test by setting high `text_threshold`).

## Any specific deployment considerations

N/A

## Docs

-   ✅  Docs updated? What were the changes: Added `box_threshold` and `text_threshold` to Grounding DINO docs.

### CODE REVIEW

File: inference/core/utils/image_utils.py [hunk 1]
```diff
@@ -47,6 +47,15 @@ def load_image_rgb(value: Any, disable_preproc_auto_orient: bool = False) -> np.
     return np_image
 
 
+def load_image_bgr(value: Any, disable_preproc_auto_orient: bool = False) -> np.ndarray:
+    np_image, is_bgr = load_image(
+        value=value, disable_preproc_auto_orient=disable_preproc_auto_orient
+    )
+    if not is_bgr:
+        np_image = cv2.cvtColor(np_image, cv2.COLOR_BGR2RGB)
```
Comment (hansent) [inline inference/core/utils/image_utils.py:55]: what is this function supposed to do?

`cv2.COLOR_BGR2RGB` sounds like its converting it from BGR to RGB,
but the if sounds like this is only being done if the if image is BGR already?

**Review by probicheaux — APPROVED** [review]
lgtm

File: inference/core/utils/image_utils.py (hunk 1 above)
Comment (probicheaux) [reply inference/core/utils/image_utils.py:55]: BGR2RGB just transposes the final channels (same as `np_image = np_image[:, :, ::-1]`)

so in the event that the returned image is not bgr, we transpose the final channels, making it bgr

File: inference/models/grounding_dino/grounding_dino.py [hunk 2]
```diff
@@ -109,12 +115,17 @@ def infer(
...
             classes=text,
-            box_threshold=0.5,
-            text_threshold=0.5,
+            box_threshold=box_threshold,
+            text_threshold=text_threshold,
         )
 
         self.class_names = text
 
+        if CLASS_AGNOSTIC_NMS:
```
Comment (hansent) [inline inference/models/grounding_dino/grounding_dino.py:124]: should this be a request based param like `box_threshold` and `text_threshold`?

Comment (skylargivens) [reply inference/models/grounding_dino/grounding_dino.py:124]: I agree. Will update this to mimic the YOLO World implementation:
https://github.com/roboflow/inference/blob/2185ed72f094345ee43b39737716a578fd051908/inference/models/yolo_world/yolo_world.py#L89

**Pull request merged**